Client/Frontend/Brave Today/Composer/FeedDataSource+RSS.swift

```diff
@@ -0,0 +1,212 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import FeedKit
+import Fuzi
+import Shared
+
+struct RSSFeedLocation: Hashable {
+    var title: String?
+    var url: URL
+    
+    var id: String {
+        url.absoluteString
+    }
+}
+
+extension FeedDataSource {
+    
+    // MARK: - RSS Sources
+    
+    var rssFeedLocations: [RSSFeedLocation] {
+        RSSFeedSource.all().compactMap {
+            guard let url = URL(string: $0.feedUrl) else { return nil }
+            return RSSFeedLocation(title: $0.title, url: url)
+        }
+    }
+    
+    /// Add a users custom RSS feed to the list of sources
+    ///
+    /// - returns: `true` if the feed is successfully added, `false` if it already exists
+    @discardableResult
+    func addRSSFeedLocation(_ location: RSSFeedLocation) -> Bool {
+        let feedUrl = location.url.absoluteString
+        if RSSFeedSource.get(with: feedUrl) != nil {
+            return false
+        }
+        RSSFeedSource.insert(title: location.title,
```

Comment:
RSS feeds are read-only, what do you think about showing a warning to the user while adding an insecure rss-feed?

Reply:
FYI this was added, as seen here: https://bravesoftware.slack.com/archives/CU56GUJ1H/p1614279591019000?thread_ts=1614191458.013000&cid=CU56GUJ1H